### PR TITLE
Universal/deferred/coat forward fallback

### PR DESCRIPTION
**Pull request by eh-unity**
### Purpose of this PR
Implement clear coat for the deferred renderer as fallback to forward rendering.

* Removed ForwardOnly from default list of object drawing to decouple actual forward renderer and deferred forward fallback from each other.
* Renamed ForwardOnly to ForwardFallback to be a bit more clear.
* Added a function to switch material to be rendered as forward fallback.
* Separated BakedLit hlsl shader code into it's own .hlsl to simplify pass descriptions.

---
### Testing status

**Manual Tests**: What did you do?
- [x] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other: 

### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [ ] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR.
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.universal/Runtime/DeferredRenderer.cs [hunk 1]
```diff
@@ -117,7 +117,7 @@ public DeferredRenderer(DeferredRendererData data) : base(data)
             // - If a material cannot be lit in deferred (unlit, bakedLit, special material such as hair, skin shader), then it should declare UniversalForwardOnly pass
```
Comment (kaychang-unity) [inline com.unity.render-pipelines.universal/Runtime/DeferredRenderer.cs:117]: This comment should be updated too.

File: com.unity.render-pipelines.universal/Editor/ShaderGUI/ShadingModels/LitGUI.cs [hunk 2]
```diff
@@ -230,6 +230,22 @@ public static SmoothnessMapChannel GetSmoothnessMapChannel(Material material)
...
+                material.SetShaderPassEnabled( "UniversalGBuffer", false);
+                material.SetShaderPassEnabled( "UniversalForwardFallback", true);
+            }
+            else
+            {
+                material.SetShaderPassEnabled( "UniversalGBuffer", true);
+                material.SetShaderPassEnabled( "UniversalForwardFallback", false);
+            }
+
+            //Debug.Log("Fallback: " + enable + " UniversalGBuffer: " + material.GetShaderPassEnabled("UniversalGBuffer") + " UniversalForwardFallback: " + material.GetShaderPassEnabled("UniversalForwardFallback"));
```
Comment (Verasl) [inline com.unity.render-pipelines.universal/Editor/ShaderGUI/ShadingModels/LitGUI.cs:246]: Probably should remove this commented debug

File: com.unity.render-pipelines.universal/Shaders/Lit.shader [hunk 3]
```diff
@@ -229,6 +229,69 @@ Shader "Universal Render Pipeline/Lit"
...
+            Name "ForwardLit Fallback"
+            Tags{"LightMode" = "UniversalForwardFallback"}
+
+            Blend[_SrcBlend][_DstBlend]
+            ZWrite[_ZWrite]
+            Cull[_Cull]
+
+            HLSLPROGRAM
+            #pragma exclude_renderers d3d11_9x gles
+            #pragma target 4.5
```
Comment (Verasl) [inline com.unity.render-pipelines.universal/Shaders/Lit.shader:247]: Should the forward fallback be limited target 4.5?

**Comment by eh-unity** [comment]
Discarded approach for integrating clear coat with deferred renderer. Closing.

**Pull request closed without merging**